src/Bindings/LuaState.h

```diff
@@ -112,6 +115,63 @@ class cLuaState
 	} ;
 	
 	
+	/** Represents a callback to Lua that C++ code can call.
+	Is thread-safe and unload-safe.
+	When the Lua state is unloaded, the callback returns an error instead of calling into non-existent code.
+	To receive the callback instance from the Lua side, use RefStack() or (better) cLuaState::GetStackValue(). */
+	class cCallback
+	{
+	public:
+		/** Creates an unbound callback instance. */
+		cCallback(void) = default;
+
+		~cCallback()
+		{
+			Clear();
+		}
+
+		/** Calls the Lua callback, if still available.
+		Returns true if callback has been called.
+		Returns false if the Lua state isn't valid anymore. */
+		template <typename... Args>
+		bool Call(Args &&... args)
```

Comment:
The best way of thinking about lambdas is that they are syntax sugar. Each lambda is converted to an anonymous class, with members for each capture, and an `operator()` for the code. So 
```
auto x = [=foo, &bar](int baz) {return foo + bar + baz; };
```
is equivalent to:
```
class _AJFLKD
{
public:
    _AJFLKD(decltype(foo) a_foo, decltype(bar)& a_bar) : m_foo(a_foo), m_bar(a_bar) {}
    int operator()(int baz)
    {
        return m_foo + m_bar + baz;
    }
private:
    decltype(foo) m_foo;
    decltype(bar)& m_bar;
}
```

Admittedly some of the mistakes with references are slightly easier to make with that manual functors, but they are just as possible.